src/macros.jl

```diff
@@ -130,6 +137,26 @@ end
 
 getloopedcode(c, code, condition, idxvars, idxsets, idxpairs, sym) = code
 
+localvar(x::Expr) = Expr(:block, _localvar(x)...)
+_localvar(x::Symbol) = :(local $(esc(x)))
+function _localvar(x::Expr)
+    @assert x.head == :escape
+    args = Any[]
+    for t in x.args
+        if isa(t, Symbol)
+            push!(args, :(local $(esc(t))))
+        else
+            @assert isa(t, Expr)
+            if t.head in (:tuple, :vect, :vcat, :hcat, :row)
```

Comment:
``(i,j)``, ``[i,j]``, ``[i;j]``, ``[i j]``, and ``[i j; k l]`` (for the last two)